account_constraints/view/account_bank_statement.xml

```diff
@@ -13,7 +13,8 @@
             <field name="arch" type="xml">
                 <xpath expr="//field[@name='bank_account_id']" position="after">
                     <field name="state" invisible="1"/>
-                    <button name="cancel" attrs="{'invisible': ['|', ('journal_entry_id', '=', False), ('state', '=', 'confirm')]}" string="Cancel" type="object" icon="gtk-undo"/>
+                    <field name='account_cancel_installed' invisible="1"/>
+                    <button name="cancel" attrs="{'invisible': ['|', ('journal_entry_id', '=', False),('account_cancel_installed','=',True) ,('state', '=', 'confirm')]}" string="Cancel" type="object" icon="gtk-undo"/>
```

Comment:
I think the domain should be:
```
['|', '|', ('journal_entry_id', '=', False), ('account_cancel_installed', '=', True) ,('state', '=', 'confirm')]
```
Your domain has a `AND` between the 2 last conditions.